common/src/main/java/org/keycloak/common/util/BouncyIntegration.java

```diff
@@ -31,34 +32,20 @@ public class BouncyIntegration {
 
     private static final Logger log = Logger.getLogger(BouncyIntegration.class);
 
-    private static final String[] providerClassNames = {
-            "org.bouncycastle.jce.provider.BouncyCastleProvider",
-            "org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider"
-    };
-
     public static final String PROVIDER = loadProvider();
 
     private static String loadProvider() {
-        for (String providerClassName : providerClassNames) {
-            try {
-                Class<?> providerClass = Class.forName(providerClassName, true, BouncyIntegration.class.getClassLoader());
-                Constructor<Provider> constructor = (Constructor<Provider>) providerClass.getConstructor();
-                Provider provider = constructor.newInstance();
-
-                if (Security.getProvider(provider.getName()) == null) {
-                    Security.addProvider(provider);
-                    log.debugv("Loaded {0} security provider", providerClassName);
-                } else {
-                    log.debugv("Security provider {0} already loaded", providerClassName);
-                }
-
-                return provider.getName();
-            } catch (Exception e) {
-                log.debugv("Failed to load {0}", e, providerClassName);
-            }
+        Provider provider = CryptoIntegration.getProvider().getCryptoUtility(Provider.class, CryptoProviderTypes.BC_SECURITY_PROVIDER);
```

Comment:
Based on my proposal, this line would be `Provider provider = CryptoIntegration.getProvider().getFipsProvider();`.

Reply:
In that case, I probably did not understand your previous proposal, which talks about `getAlgorithmProvider` :-)

Could you please elaborate a bit more how would the method `getAlgorithmProvider` look like for your proposal and how would the callers look like?

If you prefer the specific methods for this use-case, it is cleaner to me to have something like `Provider provider = CryptoIntegration.getProvider().getBouncyProvider()` rather than `getFipsProvider` as it is really BouncyCastle security provider returned here. WDYT?

Reply:
Ah, ok :) Thanks for the clarification